runtime/metricsview/astsql.go

```diff
@@ -104,8 +117,21 @@ func (b *sqlBuilder) writeSelect(n *SelectNode) error {
 
 		// Add unnest joins. We only and always apply these against FromPlain (ensuring they are already unnested when referenced in outer SELECTs).
 		for _, u := range n.Unnests {
-			b.out.WriteString(", ")
-			b.out.WriteString(u)
+			if n.JoinComparisonSelect != nil {
+				b.out.WriteString("JOIN ")
+				b.out.WriteString(u)
+				b.out.WriteString(" ON TRUE")
+			} else {
+				b.out.WriteString(", ")
+				b.out.WriteString(u)
+			}
+		}
+
+		if n.JoinComparisonSelect != nil {
+			err := b.writeJoin(n.JoinComparisonType, n.CTEs[0], n.JoinComparisonSelect, nil)
```

Comment:
Feels unsafe to reference `n.CTEs[0]` like this – maybe in the future, another rewriter adds to `CTEs`.